Add tests for model attribute comparison in couch-to-sql toolchain (#31790)
dimagi/commcare-hq: docs/couch_to_sql_models.rst

    - Some docs have attributes that are couch ids of other docs. These are weak spots easy to forget when the referenced doc type is migrated. Add a comment so these show up in a grep for the referenced doc type.

  - Run ``makemigrations``
  - Add the test that was generated to it's respective place.
    - The test file uses `ModelAttrEquality` util which has mehods for running the equality tests.
    - The test class that is generated will have three methods namely `_couch_only_attrs`,  `_sql_only_attrs` and `test_have_same_attrs`.
    - Generally during a migration some attributes and methods are renamed or removed as per need. To accomodate the changes you can update `_couch_only_attrs` and `_sql_only_attrs`.
    - `_couch_only_attrs` is supposed to return a set of attributes and methods which are either removed, renamed or not used anymore in SQL.

>     - `_sql_only_attrs` would return a set of attributes and methods that are added newly in the SQL model.

Reviewer: ```suggestion
    - `_sql_only_attrs` would return a set of attributes and methods that are new in the SQL model.
```

>     - `test_have_same_attrs` is the test method that would run to test the equality of the attributes. The default implementation should work if you have populated above methods but you can modify it's implementation as per your requirement.

Reviewer: ```suggestion
    - `test_have_same_attrs` will test the equality of the attributes. The default implementation should work if you have populated the above methods but you can modify it's implementation as needed.
```
